app/instrumentation-tests/src/org/commcare/androidTests/MobilePinTests.kt

```diff
@@ -0,0 +1,237 @@
+        assertTrue(onView(withHint("Password")).isPresent())
+        onView(withHint("Password")).perform(typeText("123"))
+        onView(withId(R.id.login_button))
+            .perform(click())
+        assertTrue(onView(withText("Reset your PIN?")).isPresent())
+        verifyCreatePINWindow()
+
+        onView(withText("NO, BUT ASK AGAIN ON NEXT LOGIN")).perform(click())
+        InstrumentationUtility.logout()
+        assertTrue(onView(withHint("PIN")).isPresent())
+        InstrumentationUtility.openOptionsMenu()
+        onView(withText("Forgot PIN?"))
+            .perform(click())
+        assertTrue(onView(withHint("Password")).isPresent())
+        onView(withHint("Password")).perform(typeText("123"))
+        onView(withId(R.id.login_button))
+            .perform(click())
+        assertTrue(onView(withText("Reset your PIN?")).isPresent())
+        verifyCreatePINWindow()
+        onView(withText("NO, BUT ASK AGAIN ON NEXT LOGIN")).perform(click())
+        InstrumentationUtility.logout()
+
+    }
+
+
+    fun testMobilePINForMultipleApps() {
+        InstrumentationUtility.openOptionsMenu()
+        onView(withText("Go To App Manager"))
+            .perform(click())
+        onView(withId(R.id.install_app_button))
+            .perform(click())
+        InstrumentationUtility.installApp(CCZ_NAME_2)
+        pressBack()
+        onView(withId(R.id.app_selection_spinner)).perform(click())
+        onView(withText(APP_NAME_2)).perform(click())
+        InstrumentationUtility.login("test1", "123")
+        enablePIN()
+        InstrumentationUtility.login("test1", "123")
+        onView(withText("I'LL UPDATE LATER")).perform(click())
+        InstrumentationUtility.logout()
+        InstrumentationUtility.login("test1", "123")
+        assertTrue(onView(withText("Create a PIN?")).isPresent())
+        setPIN("1234")
+        InstrumentationUtility.logout()
+
+        uninstallApp(APP_NAME_1)
+        pressBack()
+        InstrumentationUtility.login("test1", "1234")
+        onView(withText("I'LL UPDATE LATER")).perform(click())
+        InstrumentationUtility.logout()
+        InstrumentationUtility.login("test1", "1234")
+        assertTrue(onView(withText("Start")).isPresent())
+
+
+    }
+
+    private fun verifyCreatePINWindow() {
+        assertTrue(onView(withText("YES, SET MY PIN NOW")).isPresent())
+        assertTrue(onView(withText("NO, BUT ASK AGAIN ON NEXT LOGIN")).isPresent())
+        assertTrue(onView(withText("NO, AND DON'T ASK AGAIN")).isPresent())
+        assertTrue(onView(withId(R.id.extra_info_button)).isPresent())
+    }
+
+
+    private fun enablePIN() {
+        InstrumentationUtility.enableDeveloperMode()
+        InstrumentationUtility.openOptionsMenu()
+        onView(withText("Settings"))
+            .perform(click())
+        onView(withText("Developer Options"))
+            .perform(click())
+        InstrumentationUtility.openOptionsMenu()
+        onView(withText("Mobile Privileges"))
```

Comment:
since this is not actually going to grant any priviliges, would not steps below fail to enable pin option ?